src/main.rs

```diff
@@ -668,10 +668,11 @@ impl App {
     fn process_backend_notification(&mut self, notification: BackendNotification) {
         match notification {
             // TODO: Render progress
-            BackendNotification::Loading { step, progress: _ } => {
+            BackendNotification::Loading { step, progress } => {
                 self.current_view = View::Loading;
                 self.status_bar_notification = StatusBarNotification::None;
-                self.loading_view.emit(LoadingInput::BackendLoading(step));
+                self.loading_view
+                    .emit(LoadingInput::BackendLoading((step, progress)));
```

Comment:
Let's use named fields like `BackendNotification::Loading` above instead of a tuple and remove TODO above that this PR is resolving.